### DISCUSSION TITLE
Why does the install flag has no confirmation like the remove flag from am?

### DISCUSSION THREAD

**Discussion by Martin15135215** [ivan-hc/AM#77] [General]
Why does the installation flag have no confirmation like the remove flag from am? In other package managers, there is an extra `-y` to signal that it should be installed without confirmation. However, with `am -i [PROGRAMM]` that it not the case. But creating a confirmation is possible, see `am -r`

**Comment by ivan-hc** [comment]
- Brief answer: to save time
- Long answer: installing something is an action you want to perform because you want it. Before you install anything you have the time to learn what that app do, how much big is the size of the file and every other information from the option `-a` or from the URL provided with this option. About the `-r` option, sometime the app you have installed can take with it your configurations (for example, after you have used the option `-H`, or if you have installed a custom profile for firefox the apps starting with the suffix "`ffwa-`", in the list). To use the option `-r` (the "r" is lowercased) is quick and you have time to decide if your action is right or not. On the contrary, if you have decided to remove it quickly, use the option `-R` (with the "R" uppercased, also you need to press SHIFT to do this, so you must think about this) to save time. However, if for any reason you have decided to interrupt the installation of an app, just CTRL+C to abort the installation (or the whole process), so the confirmation of an installation is useless in this case. Normally in APT if the package have not dependences, it installs something without asking, while if the package have one or more packages to install with it, it asks you to perform the installation. What AM/AppMan installs are standalone apps (AppImages or TAR/ZIP archives), so no other dependences are needed from my package managers. If you want a confirmation before you install anything, just perform `am -i $PROGRAM` in the first session of the terminal you use for it, and it will be shown a prompt for the password (the `sudo` command), is it not a time you have to decide if it is worth to install something or not? OK, AppMan have not this, however to confirm an installation for whitch you have spent time to search in the list (`-q`), for whitch you have learned more about it (`-a`) and for whitch you have wrote the whole argument after the command (`am -i $PROGRAM $ANOTHERPROGRAM $ANOTHERPROGRAMWITHALONGERNAMETHATDAMNITWHYIHAVEDECIDEDTONAMEITLIKETHIS`)... to add a confirmation message is an additiona waste of time inthis case, I think.

**Reply by Martin15135215** [reply]
Thanks for the explanation! I also have another question. I only needed sudo to install `am` from the GitHub repo, however afterwards for `am -i obsidian` for example, it did the installation successfully without sudo. So my question is, is sudo only needed for the first command to install `am` → `wget https://raw.githubusercontent.com/ivan-hc/AM-application-manager/main/INSTALL && chmod a+x ./INSTALL && sudo ./INSTALL` and afterwards not?

**Reply by ivan-hc** [reply]
`sudo` is included in the options `-i`, `-r`, `-R`, `--enable-completion` and `--disable-completion`, the reason is that "AM" can operate at system's level (the contrary of AppMan). If you are in the same session (and in a brief arch of time) your system's keyring will save the last time you have used the password. I think that if you use the same terminal's session after 5-10 minutes, the terminal should prompt again for the password. If you close it and open a new one, you have a new session, so use the password again. Other commands like `apt` and `pacman` require `sudo`, but an AUR helper as `yay` or `paru` don't require it if you have already used `sudo` in the same session (just like "AM").
